pydra/engine/tests/test_shelltask.py

```diff
@@ -2927,6 +2927,116 @@ def test_shell_cmd_outputspec_7b_error():
     assert "has to have a callable" in str(e.value)
 
 
+@pytest.mark.parametrize("results_function", [result_no_submitter, result_submitter])
+def test_shell_cmd_outputspec_7c(tmpdir, plugin, results_function):
+    """
+        customised output_spec, adding Directory to the output named by args
+    """
+
+    def get_lowest_directory(directory_path):
+        return str(directory_path).replace(str(Path(directory_path).parents[0]), "")
+
+    cmd = "mkdir"
+    args = [f"{tmpdir}/dir1", f"{tmpdir}/dir2"]
+
+    my_output_spec = SpecInfo(
+        name="Output",
+        fields=[
+            (
+                "resultsDir",
+                attr.ib(
+                    type=File,
```

Comment:
i believe you wanted to test `Directory` here

Reply:
That is correct. I've updated to use Directory instead of File as the output type.